opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/NonDominatedSorting.java

```diff
@@ -0,0 +1,179 @@
+package org.opt4j.optimizers.ea;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.opt4j.core.Individual;
+import org.opt4j.core.Objective;
+import org.opt4j.core.Objectives;
+
+/**
+ * Class used for the non-dominated sorting. During non-dominated sorting, the
+ * evaluated individuals are sorted into fronts based on the number of
+ * individuals they are dominated by. The first front consists of points that
+ * are not dominated at all and so on.
+ * 
+ * @author Fedor Smirnov
+ *
+ */
+public class NonDominatedSorting {
+
+	private NonDominatedSorting() {
```

Comment:
delete unused constructor

Reply:
This is a thing I got from SonarQube. It says that it is good practice to have a private constructor in classes that only offer static methods, so that no instance of the class can be created.